### PR TITLE
Migrate DeckNameComparatorTest to Kotlin

### PR DESCRIPTION
**Pull request by oyeraghib**
## Purpose / Description
Migrate DeckNameComparatorTest to Kotlin

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt [hunk 1]
```diff
@@ -1,29 +1,26 @@
...
     @Test
-    public void sortDeckNames() {
-        String[] deckNames = new String[]{"AA", "ab", "BB", "aa", "aa::bb", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB"};
-        sort(deckNames, mDeckNameComparator);
-
-        assertThat(deckNames, is(new String[]{"AA", "aa", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB", "aa::bb", "ab", "BB"}));
+    fun sortDeckNames() {
+        val deckNames = arrayOf("AA", "ab", "BB", "aa", "aa::bb", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB")
+        Arrays.sort(deckNames, mDeckNameComparator)
+        MatcherAssert.assertThat(deckNames, CoreMatchers.`is`(arrayOf("AA", "aa", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB", "aa::bb", "ab", "BB")))
```
Comment (mikehardy) [inline AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt:24]: kotlincleanup is -> equalTo

**Review by mikehardy — CHANGES_REQUESTED** [review]
Thanks!
one kotlincleanup, same as first two :-)

File: AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-    public void sortDeckNames() {
-        String[] deckNames = new String[]{"AA", "ab", "BB", "aa", "aa::bb", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB"};
-        sort(deckNames, mDeckNameComparator);
-
-        assertThat(deckNames, is(new String[]{"AA", "aa", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB", "aa::bb", "ab", "BB"}));
+    @KotlinCleanup("is --> equalTo")
+    fun sortDeckNames() {
+        val deckNames = arrayOf("AA", "ab", "BB", "aa", "aa::bb", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB")
+        Arrays.sort(deckNames, mDeckNameComparator)
+        MatcherAssert.assertThat(deckNames, CoreMatchers.`is`(arrayOf("AA", "aa", "aa::ab", "aa::ab::Aa", "aa::ab::aB", "aa::ab:bB", "aa::bb", "ab", "BB")))
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt:25]: import: `MatcherAssert` & `CoreMatchers`

**Review by david-allison — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/test/java/com/ichi2/utils/DeckNameComparatorTest.kt:25]: Ah - nice, that makes the code match almost exactly before vs after, just the import statement different but those are easy to review vs code movement :+1:

**Review by mikehardy — APPROVED** [review]
agree with David on the imports (and I learned them better, for it)
Once those are handled I approve, so just blocking on that

**Review by david-allison — APPROVED** [review]

**Pull request merged**